app/src/main/java/de/xikolo/storages/RecentCoursesStorage.kt

```diff
@@ -0,0 +1,45 @@
+package de.xikolo.storages
+
+import android.content.Context
+import com.google.gson.Gson
+import com.google.gson.reflect.TypeToken
+import de.xikolo.storages.base.BaseStorage
+
+typealias RecentCourse = Pair<String, String>
+
+const val MAX_SHORTCUTS = 4
+
+class RecentCoursesStorage : BaseStorage(PREF_RECENT_COURSES, Context.MODE_PRIVATE) {
+
+    val recentCourses: LinkedHashSet<RecentCourse>
+        get() {
+            val json = getString(PREF_RECENT_COURSES)
+            val type = object : TypeToken<LinkedHashSet<RecentCourse>>() {}.type
+            if (json == null || type == null) {
+                return LinkedHashSet<RecentCourse>()
+            }
+            return Gson().fromJson(json, type)
+        }
+
+    fun addCourse(courseId: String, title: String) {
+        var courses: LinkedHashSet<RecentCourse> = recentCourses
+
+        courses = courses
+            // filter duplicates
+            .filter { it.first != courseId }
+            .toMutableSet()
```

Comment:
Why is it not mutable in the first place?

Reply:
filter returns a non-mutable List, therefore I had to make it a mutableSet to use the add method in the following line.